sdk/core/azure-core/src/operation_status.cpp

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "azure/core/operation_status.hpp"
+
+namespace Azure { namespace Core {
+
+  const OperationStatus OperationStatus::NotStarted("NotStarted");
+  const OperationStatus OperationStatus::Running{"Running"};
+  const OperationStatus OperationStatus::Succeeded{"Succeeded"};
+  const OperationStatus OperationStatus::Failed{"Failed"};
+  const OperationStatus OperationStatus::Cancelled{"Cancelled"};
```

Comment:
Be consistent when using `()` vs `{}`

Also, out of curiosity, why are the enum values in the cpp file and not in the header? I see the guidelines tell us to do it this way, but want to understand why:
https://azure.github.io/azure-sdk/cpp_design.html#enumerations
@CaseyCarter?

Reply:
@ahsonkhan only the definition is in the header.  The initialization is done on source files.
It's similar to function signatures.

Reply:
I understand we do that generally, but sometimes, especially for functions with small implementations, we end up writing it in the header file directly. We are even doing that in this PR, for example:
https://github.com/Azure/azure-sdk-for-cpp/blob/3fea31fee2d549b7070732a06d23b40c7aa52206/sdk/core/azure-core/inc/azure/core/operation.hpp#L29-L34
https://github.com/Azure/azure-sdk-for-cpp/blob/3fea31fee2d549b7070732a06d23b40c7aa52206/sdk/core/azure-core/inc/azure/core/operation_status.hpp#L25-L33

So, why not do the same for enums. It looks like the guidelines specifically went out of the way to call out and suggest providing the initialization of these enum values in the cpp file. Why?

When should we leave the implementation in the header file versus move to the source?

Reply:
> When should we leave the implementation in the header file versus move to the source?

For functions:  Only when the implementation is kind of short (a few lines) and does not require/use external libraries which will require including an external header (like using json, or something).
It is usually for set/get method implementation where there's just a return or something simple.

For variables (like this case): Only when the variable is a `constexpr`, which means we can assign a constant value duting the definition. This is true for non Heap-allocated stuff (int, enums, char*, etc). But when it is a reference to a Class (like this example), we can't create a constant object of the class statically.  So we just define the symbol and assign the value in source